src/debugger/appWorker.ts

```diff
@@ -186,8 +189,12 @@ export class MultipleLifetimesAppWorker {
         socketToApp.on("message",
             (message: any) => this.onMessage(message));
         socketToApp.on("error",
-            (error: Error) => printDebuggingError("An error ocurred while using the socket to communicate with the React Native app", error));
-        return socketToApp;
+            (error: Error) => {
+                deferred.reject(error);
+            });
+
+        Q.delay(300).done(() => deferred.resolve(socketToApp));
```

Comment:
Should leave a comment as to why the 300 is necessary. Also - if we're using 300 in other places would be good to define a websocket timeout constant for everyone to use (not necessarily with this change though).